docs/HOL_Deploy_Private_Docker_Registry/README.md

```diff
@@ -0,0 +1,447 @@
+```
+$ sudo mkdir /usr/local/share/ca-certificates/docker-dev-cert
+$ sudo cp domain.crt /usr/local/share/ca-certificates/docker-dev-cert
+$ sudo update-ca-certificates
+```
+
+**Step 10.** Restart the Docker daemon so that it picks up the changes made in certificate store:
+```
+$ sudo service docker restart
+```
+
+**Step 11.** It's time to start up an instance of your private Docker registry:
+```
+$ cd ~/docker-registry
+$ docker-compose up
+```
+
+**Step 12.** Open up another terminal and verify that the private Docker registry is running by executing:
+```
+$ docker ps
+```
+The output should look similar to the following:
+```
+CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS                          NAMES
+d4b6fef0b4d1        nginx:1.9           "nginx -g 'daemon of   2 minutes ago       Up 2 minutes        80/tcp, 0.0.0.0:443->443/tcp   dockerregistry_nginx_1
+77668352bd39        registry:2          "registry cmd/regist   2 minutes ago       Up 2 minutes        127.0.0.1:5000->5000/tcp       dockerregistry_registry_1
+```
+> **Note:** The names all start with dockerregistry_
+
+**Step 13.** Let's try another curl test using your credentials and domain name:
+```
+$ curl https://<user-name>:<password>@<domain-name>/v2/
+```
+As of this writing Docker returns an empty json object, so you should see:
+```
+{}
+```
+
+**Step 14.** Log in to Docker registry from a client machine by executing:
+```
+$ docker login https://<domain-name>
+```
+Enter the username and password you created earlier, and you should see the following message:
+```
+Login Succeeded
+```
+> **Note:** If you use a self-signed certificate, please add the SSL certificate you created in Step 14 to this client machine.
+
+**Step 15.** You are now ready to publish images to the private Docker registry. Create Docker images by following this [Dockerizing Parts Unlimited MRP](https://microsoft.github.io/PartsUnlimitedMRP/adv/adv-21-Docker.html) lab. Tag an image with your registry's domain name (e.g., myregistrydomain.com/web) in order to push it:
+```
+$ docker tag mypartsunlimitedmrp/web <domain-name>/web
+```
+
+**Step 16.** Now we can push this image (e.g., myregistrydomain.com/web) to the private Docker registry:
+```
+$ docker push <domain-name>/web
+```
+This will take a moment to upload to the registry server. You should see output that ends with something similar to the following:
+```
+latest: digest: sha256:d35c98ab2f3ec8d963e0d20a3577c8da23f7366d129959195749858da964fb0d size: 3251
+```
+New Docker image appears in your Azure Blob storage.
+
+![](<media/repo.png>)
+
+**Step 17.** Pull this image (e.g., myregistrydomain.com/web) back from private Docker registry:
+```
+$ docker pull <domain-name>/web
+```
+
+###Task 2: Set up Continuous Integration (CI) with Visual Studio Team Services (VSTS)  
+
```

Comment:
Please add an intro / context on what we gonna do during this task. Introduce Docker-Bench and Docker Inspect command